### PR TITLE
fix: When users manually selects failed workflow - it is truncated in a weird wa

### PR DESCRIPTION
**Pull request by eugenejahn**
# TL;DR

fixed by updating the state and recalculating row size 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
when the rows update(adding or removing item), recalculate the size and cache it in the state. 

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyteconsole/issues/386

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by anrusina — APPROVED** [review]
LGTM,
